twinify/dataframe_data.py

```diff
@@ -27,13 +27,37 @@
 Dtype = TypeVar("Dtype")
 
 
+def integers_to_categories_handler(col: pd.Series) -> Dtype:
+    assert is_integer_dtype(col.dtype)
+    return pd.CategoricalDtype(np.unique(col))
+
+
+def integers_to_range_handler(col: pd.Series) -> Dtype:
+    assert is_integer_dtype(col.dtype)
+    return pd.CategoricalDtype(np.arange(col.min(), col.max() + 1))
+
+
+def keep_integers_handler(col: pd.Series) -> Dtype:
+    assert is_integer_dtype(col.dtype)
+    return col.dtype
+
+
+def disallow_integers(col: pd.Series) -> Dtype:
+    assert is_integer_dtype(col.dtype)
+    raise ValueError(f"Columns with pure integers are not allowed (but {col.name} was); make sure all columns are of type categorical.")
+
+
 class DataDescription:
 
     def __init__(self, dtypes: Mapping[str, Dtype]) -> None:
         self._dtypes = dict(dtypes)
 
     @staticmethod
-    def from_dataframe(df: pd.DataFrame, strings_to_categories: bool = True) -> "DataDescription":
+    def from_dataframe(
+            df: pd.DataFrame, 
+            strings_to_categories: bool = True, 
+            integers_handler: Callable[[pd.Series], Dtype] = keep_integers_handler,
```

Comment:
If the default handler is `disallow_integers` in NapsuMQModel, shouldn't it be the same here?

Reply:
`DataDescription` is used by the DPVI code as well. Since integers are fine there, I decided to allow them by default in `DataDescription.from_dataframe` (which then gets overwritten by the default in `NapsuMQModel` if called from there).